Version 1.3 errors with "tsc".

# Context
- **OS and version used:** Windows 10
- **Node.js version:** 8.3
- **npm version:** 5.3
- **list of installed packages:** 
+-- @types/node@8.5.7
+-- azure-iot-device@1.3.0
+-- azure-iot-device-amqp@1.3.0
`-- typescript@2.6.2
- **cloned repo:** https://github.com/robertsimmons/azure-iot-sdk-tsc-bug

# Description of the issue:
Noticed this is my main project after upgrading azure-iot-device and azure-iot-device-amqp, but created a new repo with just those packages and typescript, and get the same result. 

# Code sample exhibiting the issue:
https://github.com/robertsimmons/azure-iot-sdk-tsc-bug

Just clone that, run "npm i", "npm run tsc --init", then "npm run build".

# Console log of the issue:
npm run build

> test@1.0.0 build C:\src\azure-iot-sdk-tsc-bug
> npm run tsc


> test@1.0.0 tsc C:\src\azure-iot-sdk-tsc-bug
> tsc

node_modules/azure-iot-device/lib/x509_authentication_provider.d.ts(9,22): error TS2420: Class 'X509AuthenticationProvider' incorrectly implements interface 'AuthenticationProvider'.
  Types of property 'getDeviceCredentials' are incompatible.
    Type '(callback: (err: Error, credentials?: TransportConfig | undefined) => void) => void' is not assignable to type '(callback: (err: Error, credentials: TransportConfig) => void) => void'.
      Types of parameters 'callback' and 'callback' are incompatible.
        Types of parameters 'credentials' and 'credentials' are incompatible.
          Type 'TransportConfig | undefined' is not assignable to type 'TransportConfig'.
            Type 'undefined' is not assignable to type 'TransportConfig'.
npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! test@1.0.0 tsc: `tsc`
npm ERR! Exit status 2
npm ERR!
npm ERR! Failed at the test@1.0.0 tsc script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\rob\AppData\Roaming\npm-cache\_logs\2018-01-08T14_56_05_016Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! test@1.0.0 build: `npm run tsc`
npm ERR! Exit status 2
npm ERR!
npm ERR! Failed at the test@1.0.0 build script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\rob\AppData\Roaming\npm-cache\_logs\2018-01-08T14_56_05_052Z-debug.log

--- a/common/core/src/authentication_provider.ts
+++ b/common/core/src/authentication_provider.ts
@@ -23,5 +23,5 @@ export enum AuthenticationType {
  */
 export interface AuthenticationProvider {
   type: AuthenticationType;
-  getDeviceCredentials(callback: (err: Error, credentials: TransportConfig) => void): void;
+  getDeviceCredentials(callback: (err: Error, credentials?: TransportConfig) => void): void;
 }

--- a/device/core/src/sak_authentication_provider.ts
+++ b/device/core/src/sak_authentication_provider.ts
@@ -53,7 +53,7 @@ export class SharedAccessKeyAuthenticationProvider extends EventEmitter implemen
    *
    * @param callback function that will be called with either an error or a set of device credentials that can be used to authenticate with the IoT hub.
    */
-  getDeviceCredentials(callback: (err: Error, credentials: TransportConfig) => void): void {
+  getDeviceCredentials(callback: (err: Error, credentials?: TransportConfig) => void): void {
     if (this._shouldRenewToken()) {
       this._renewToken();
     }

--- a/device/core/src/sas_authentication_provider.ts
+++ b/device/core/src/sas_authentication_provider.ts
@@ -35,7 +35,7 @@ export class SharedAccessSignatureAuthenticationProvider extends EventEmitter im
    *
    * @param callback function that will be called with either an error or a set of device credentials that can be used to authenticate with the IoT hub.
    */
-  getDeviceCredentials(callback: (err: Error, credentials: TransportConfig) => void): void {
+  getDeviceCredentials(callback: (err: Error, credentials?: TransportConfig) => void): void {
     /*Codes_SRS_NODE_SAS_AUTHENTICATION_PROVIDER_16_002: [The `getDeviceCredentials` method shall call its callback with a `null` error parameter and the stored `credentials` object containing the current device credentials.]*/
     callback(null, this._credentials);
   }

